code/modules/modular_computers/file_system/programs/security/penalmechs.dm

```diff
@@ -0,0 +1,128 @@
+/datum/computer_file/program/penal_mechs
+	filename = "penalmechs"
+	filedesc = "Remote Penal Monitoring"
+	program_icon_state = "security"
+	extended_desc = "This program allows monitoring and control of active penal miner mechs."
+	required_access_run = access_armory
+	required_access_download = access_armory
+	requires_ntnet = 1
+	available_on_ntnet = 1
+	network_destination = "penal mining mech monitoring system"
+	size = 11
+	usage_flags = PROGRAM_ALL_REGULAR
+	color = LIGHT_COLOR_ORANGE
+	nanomodule_path = /datum/nano_module/camera_monitor
+
+/datum/computer_file/program/penal_mechs/ui_interact(mob/user)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+	if (!ui)
+		ui = new /datum/vueui/modularcomputer(user, src, "mcomputer-security-penalcontroller", 800, 600, "Penal Mech Monitoring")
+		ui.auto_update_content = 1
+	ui.open()
+
+/datum/computer_file/program/penal_mechs/vueui_transfer(oldobj)
+	for(var/o in SSvueui.transfer_uis(oldobj, src, "mcomputer-security-penalcontroller", 800, 600, "Penal Mech Monitoring"))
+		var/datum/vueui/ui = o
+		// Let's ensure our ui's autoupdate after transfer.
+		// TODO: revert this value on transfer out.
+		ui.auto_update_content = 1
+	return TRUE
+
+
+/datum/computer_file/program/penal_mechs/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	. = ..()
+	data = . || data || list()
+
+	// Gather data for computer header
+	var/headerdata = get_header_data(data["_PC"])
+	if(headerdata)
+		data["_PC"] = headerdata
+		. = data
+	
+	var/datum/signal/signal
+	signal = telecomms_process_active()
+
+	var/list/mechs = list()
+	var/list/robots = list()
+
+	if(signal.data["done"] == 1)
+		for(var/mech in SSvirtualreality.mechs["prisonmechs"])
+			var/mob/living/heavy_vehicle/M = mech
+
+			if(!ismech(M))
+				continue
+			if(!AreConnectedZLevels(computer.loc.z, M.loc.z))
+				continue
+			var/list/mechData = list("location" = null, "name" = null, "pilot" = null, "ref" = "\ref[M]", "camera" = null, "lockdown" = null)
+
+			mechData["name"] = M.name
+			mechData["pilot"] = "[M.old_mob]"
+			var/turf/mech_turf = get_turf(M)
+			mechData["location"] = "[mech_turf.x], [mech_turf.y], [mech_turf.z]"
+
+			mechData["camera"] = M.camera.status
+			mechData["lockdown"] = M.lockdown
+			mechs[++mechs.len] = mechData
+
+		for(var/robot in SSvirtualreality.robots["prisonrobots"])
+			var/mob/living/R = robot
+
+			if(!ismob(R))
+				continue
+			if(!AreConnectedZLevels(computer.loc.z, R.loc.z))
+				continue
+			var/list/robotData = list("location" = null, "name" = null, "pilot" = null, "ref" = "\ref[R]")
+
+			robotData["name"] = R.name
+			robotData["pilot"] = "[R.old_mob]"
+			var/turf/robot_turf = get_turf(R)
+			robotData["location"] = "[robot_turf.x], [robot_turf.y], [robot_turf.z]"
+
+			robots[++robots.len] = robotData
+
+	data["mechs"] = sortByKey(mechs, "pilot")
+	data["robots"] = sortByKey(robots, "pilot")
+
+	var/datum/nano_module/camera_monitor/CM = NM
```

Comment:
Use of nanomodule is smart... but a bit of a hacky way. I would appropriate if it would be done better. But I get if you want to keep using it.

_Please keep in mind that someday NanoUI might be removed_

Reply:
Removed dependency on NanoUI.